mlir/lib/Dialect/Vector/Transforms/VectorLinearize.cpp

```diff
@@ -19,10 +19,27 @@
 
 using namespace mlir;
 
+static bool isLessThanTargetBitWidth(Operation *op, unsigned targetBitWidth) {
+  auto resultTypes = op->getResultTypes();
+  for (auto resType : resultTypes) {
+    VectorType vecType = cast<VectorType>(resType);
+    unsigned trailingVecDimBitWidth =
+        vecType.getShape().back() * vecType.getElementTypeBitWidth();
```

Comment:
I think, vectors of index type are allowed, and `getElementTypeBitWidth()` will crash for them, we need to handle them somehow (probably just ignoring them entirely in this pass).

Reply:
Fixed.

Reply:
Can you please add test for index type and check that IR was left unchanged.

Reply:
Can you please give me an example of using tensor of indices? I have never seen one before

Reply:
just something like `arith.addi %a, %b : vector<2x2xindex>`

Reply:
Ah Ok. Sure, added here: [d429875](https://github.com/llvm/llvm-project/pull/83314/commits/d4298757c62bdd6415f1199bf93707b8a3b592b0)